redisinsight/api/src/modules/rdi/providers/rdi.client.storage.ts

```diff
@@ -1,54 +1,96 @@
 import { RdiClient } from 'src/modules/rdi/client/rdi.client';
-import { Injectable, Logger } from '@nestjs/common';
+import { BadRequestException, Injectable, Logger } from '@nestjs/common';
 import { RdiClientMetadata } from 'src/modules/rdi/models';
+import { sum } from 'lodash';
+import { RDI_SYNC_INTERVAL } from 'src/modules/rdi/constants';
 
 @Injectable()
 export class RdiClientStorage {
   private readonly logger = new Logger('RdiClientStorage');
 
   private readonly clients: Map<string, RdiClient> = new Map();
 
-  // todo: sync clients (idle check)
+  private readonly syncInterval: NodeJS.Timeout;
+
+  constructor() {
+    this.syncInterval = setInterval(this.syncClients.bind(this), RDI_SYNC_INTERVAL);
+  }
+
+  onModuleDestroy() {
+    clearInterval(this.syncInterval);
+  }
+
+  /**
+   * Removes all clients with exceeded idle threshold
+   * @private
+   */
+  private syncClients(): void {
+    this.clients.forEach((client) => {
+      if (client.isIdle()) {
+        this.clients.delete(client.id);
+      }
+    });
+  }
 
   async get(id: string): Promise<RdiClient> {
     const client = this.clients.get(id);
-
     if (client) {
       client.setLastUsed();
-      if (!(await client.isConnected())) {
-        await this.delete(id);
-        return null;
-      }
     }
 
     return client;
   }
 
   async getByMetadata(rdiClientMetadata: RdiClientMetadata): Promise<RdiClient> {
-    return this.clients.get(RdiClient.generateId(rdiClientMetadata));
```

Comment:
let's add validation of rdiClientMetadata here. We must check that id, sessionMetadata.userId, sessionMetadata,sessionId are defined (may be something else?)